src/Symfony/Component/DependencyInjection/LazyProxy/InheritanceProxyHelper.php

```diff
@@ -0,0 +1,141 @@
+        }
+        if ($r->isStatic()) {
+            throw new RuntimeException(sprintf('Unable to configure getter injection for service "%s": method "%s::%s" cannot be static.', $id, $class->name, $r->name));
+        }
+        if ($r->isFinal()) {
+            throw new RuntimeException(sprintf('Unable to configure getter injection for service "%s": method "%s::%s" cannot be marked as final.', $id, $class->name, $r->name));
+        }
+        if ($r->returnsReference()) {
+            throw new RuntimeException(sprintf('Unable to configure getter injection for service "%s": method "%s::%s" cannot return by reference.', $id, $class->name, $r->name));
+        }
+        if (0 < $r->getNumberOfParameters()) {
+            throw new RuntimeException(sprintf('Unable to configure getter injection for service "%s": method "%s::%s" cannot have any arguments.', $id, $class->name, $r->name));
+        }
+
+        return $r;
+    }
+
+    public static function getSignature(\ReflectionFunctionAbstract $r, &$call = null)
+    {
+        $signature = array();
+        $call = array();
+
+        foreach ($r->getParameters() as $i => $p) {
+            $k = '$'.$p->name;
+            if (method_exists($p, 'isVariadic') && $p->isVariadic()) {
+                $k = '...'.$k;
+            }
+            $call[] = $k;
+
+            if ($p->isPassedByReference()) {
+                $k = '&'.$k;
+            }
+            if (method_exists($p, 'getType')) {
+                $type = $p->getType();
+            } elseif (preg_match('/^(?:[^ ]++ ){4}([a-zA-Z_\x7F-\xFF][^ ]++)/', $p, $type)) {
+                $type = $type[1];
+            }
+            if ($type && $type = self::getTypeHint($type, $r)) {
+                $k = $type.' '.$k;
+            }
+            if ($type && $p->allowsNull()) {
+                $k = '?'.$k;
+            }
+
+            try {
+                $k .= ' = '.self::export($p->getDefaultValue());
+                if ($type && $p->allowsNull() && null === $p->getDefaultValue()) {
+                    $k = substr($k, 1);
+                }
+            } catch (\ReflectionException $e) {
+                if ($type && $p->allowsNull() && !class_exists('ReflectionNamedType', false)) {
+                    $k .= ' = null';
+                    $k = substr($k, 1);
+                }
+            }
+
+            $signature[] = $k;
+        }
+        $call = ($r->isClosure() ? '' : $r->name).'('.implode(', ', $call).')';
+
+        if ($type = method_exists($r, 'getReturnType') ? $r->getReturnType() : null) {
+            $type = ': '.($type->allowsNull() ? '?' : '').self::getTypeHint($type, $r);
+        }
+
+        return ($r->returnsReference() ? '&' : '').($r->isClosure() ? '' : $r->name).'('.implode(', ', $signature).')'.$type;
+    }
+
+    public static function getTypeHint($type, \ReflectionFunctionAbstract $r)
+    {
+        if (is_string($type)) {
```

Comment:
this case only happen on PHP 5.x, right ? If yes, we should add a comment about this, to be helpful for the guy doing the cleanup in Symfony 4. If no, we need to support ``iterable``, ``void`` and scalar types just below.